frontend/lib/src/dataframes/Quiver.ts

```diff
@@ -875,6 +876,26 @@ but was expecting \`${JSON.stringify(expectedIndexTypes)}\`.
       .format(timeInSeconds % 1 === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
   }
 
+  private static formatDecimal(value: Uint32Array, scale: number): string {
+    // This code is partly based on: https://github.com/apache/arrow/issues/35745
+    let numString = util
+      .bigNumToString(new util.BN(value))
+      .padStart(scale, "0")
+
+    if (scale === 0) {
+      return numString
+    }
+    // Arrow JS does not handle the fractional part yet:
+    let sign = ""
+    if (numString.startsWith("-")) {
+      sign = "-"
+      numString = numString.slice(1)
+    }
+    const wholePart = numString.slice(0, -scale) || "0"
```

Comment:
This is to handle if scale is essentially 0, and the slicing produces empty string. This means the decimal is purely `< 1` and thus is `0.something`.

Reply:
Almost correct, but it handles if the scale is equal to the number's total length. `scale === 0` would mean that it's only a non-fractional number.